textpipe/doc.py

```diff
@@ -542,19 +542,18 @@ def generate_gensim_document_embedding(self, model_file=None, lowercase=True):
         True
         """
         if not model_file:
-            raise TextpipeMissingModelException('No Gensim word2vec model file specified.')
+            raise TextpipeMissingModelException('No Gensim keyed vector file specified.')
         try:
             model = self._load_gensim_word2vec_model(model_file)
         except FileNotFoundError:
-            raise TextpipeMissingModelException(f'Gensim model file {model_file} is not available.')
+            raise TextpipeMissingModelException(f'Gensim keyed vector file {model_file} is not available.')
 
-        weighted_word_vectors = []
-        for word, count in self.word_counts.items():
-            word = word.lower() if lowercase else word
-            if word in model.vocab:
-                weighted_word_vectors.append(model[word] * count / model.vocab[word].count)
+        prepared_word_counts = [(word.lower() if lowercase else word, count)
+                                for word, count in self.word_counts.items()]
+
+        doc_vector = sum([model[word] * (count / model.vocab[word].count)
+                          for word, count in prepared_word_counts if word in model])
```

Comment:
cool 

(for those who like me wondered whether this matters performance-wise);
```In [39]: %timeit old(word_counts, model)
99 µs ± 4.44 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [40]: %timeit new(word_counts, model)
82.4 µs ± 1.08 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```